pkg/installer/installer.go

```diff
@@ -107,6 +107,15 @@ func createMutatingWebhookConfiguration(certificate []byte, failurePolicyStr str
 	}
 	sideEffects := arv1.SideEffectClassNone
 	path := "/mutate"
+	namespaceSelector := metav1.LabelSelector{
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      "kubernetes.io/metadata.name",
+				Operator: metav1.LabelSelectorOpNotIn,
+				Values:   []string{"kube-system"},
```

Comment:
OK, so here is what i found.

1. it is recommended to avoid operating on kube-system namespace [1]
2. it is recommended to avoid operating in the namespace the webhook is deployed to avoid deadlocks [2]

these two namespaces might not be the same (by default here they are). you should also add the namespace passed to `installer.Install()` if its different than "kube-system"

[1] https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#avoiding-operating-on-the-kube-system-namespace
[2] https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#avoiding-deadlocks-in-self-hosted-webhooks

Reply:
@VivekThrivikraman-est apart of this i think we are good.

Reply:
@adrianchiris "namespace" passed to installer seems to be used for creation of service . The namespace of webhook seems to be directly specified in server.yaml (as kube-system). Should the installer container(init container) get the namespace internally(through env variables like nri container currently does) and add that to the list ? I think that would be required if user changes the namespace in server.yaml and also the auth.yaml (to have nri pod in another namespace).

Reply:
> Should the installer container(init container) get the namespace internally(through env variables like nri container currently does) and add that to the list ?

it might, but thats not related to this change IMO. you could also pass `--namespace=${NAMESPACE}` in initcontainer args. `NAMESPACE` being an environment variable containing the pod's namespace .

my ask here is IF the user decided to change where webhook and its components deploy (server.yaml, auth.yaml)
and provided the correct value for the namespace flag  to Installer, then this bit should take it into account.

Reply:
> my ask here is IF the user decided to change where webhook and its components deploy (server.yaml, auth.yaml)
and provided the correct value for the namespace flag to Installer, then this bit should take it into account.

Ok will make that change and push.

Reply:
> @adrianchiris "namespace" passed to installer seems to be used for creation of service . The namespace of webhook seems to be directly specified in server.yaml (as kube-system). Should the installer container(init container) get the namespace internally(through env variables like nri container currently does) and add that to the list ? 

@VivekThrivikraman-est  Yes, I think it should. (could be in a separate PR)
Just for my understanding: How did you create the webhook.yaml and server.yaml? It looks to me that we would need either webhook.yaml or installer initContainer, but not both since they do the same webhook creation.

Reply:
> It looks to me that we would need either webhook.yaml or installer initContainer, but not both since they do the same webhook creation.

@zshi-redhat yes it would be only one required, I made changes so that it is consistent both ways.